enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult/InteractiveQuestionResult.tsx

```diff
@@ -0,0 +1,139 @@
+import { useDisclosure } from "@mantine/hooks";
+import { type ReactElement, type ReactNode, useState } from "react";
+import { match } from "ts-pattern";
+import { t } from "ttag";
+
+import {
+  SdkError,
+  SdkLoader,
+} from "embedding-sdk/components/private/PublicComponentWrapper";
+import { Box, Button, Group, Icon } from "metabase/ui";
+
+import { InteractiveQuestion } from "../../public/InteractiveQuestion";
+import { useInteractiveQuestionContext } from "../InteractiveQuestion/context";
+
+import InteractiveQuestionS from "./InteractiveQuestionResult.module.css";
+
+export interface InteractiveQuestionResultProps {
+  height?: string | number;
+  withResetButton?: boolean;
+  withTitle?: boolean;
+  customTitle?: ReactNode;
+}
+
+type QuestionView = "notebook" | "filter" | "summarize" | "visualization";
+
+const ContentView = ({
+  questionView,
+  onReturn,
+}: {
+  questionView: QuestionView;
+  onReturn: () => void;
+}) =>
+  match<QuestionView>(questionView)
+    .with("filter", () => <InteractiveQuestion.Filter onClose={onReturn} />)
+    .with("summarize", () => (
+      <InteractiveQuestion.Summarize onClose={onReturn} />
+    ))
+    .with("notebook", () => <InteractiveQuestion.Notebook onApply={onReturn} />)
+    .otherwise(() => <InteractiveQuestion.QuestionVisualization />);
+
+export const InteractiveQuestionResult = ({
+  height,
+  withTitle,
+  customTitle,
+  withResetButton,
+}: InteractiveQuestionResultProps): ReactElement => {
+  const [questionView, setQuestionView] =
+    useState<QuestionView>("visualization");
+
+  const { question, queryResults, isQuestionLoading } =
+    useInteractiveQuestionContext();
+
+  const [isChartSelectorOpen, { toggle: toggleChartTypeSelector }] =
+    useDisclosure(true);
```

Comment:
Should we have a prop to set the initial open/closed state for chart type sidebar? I don't think it always make sense to have the chart type selector open by default.

Reply:
Yep, I think a prop will be good here. Using `true` was just for my testing so I'll change it to false and sort the layout first before adding anything